server/src/main/java/com/google/musicanalysis/site/YoutubeServlet.java

```diff
@@ -56,54 +56,22 @@ protected String getYoutubeRes(String apiKey, String accessToken)
     }
 
     /**
-     * checks whether topic is categorized as music
-     * by checking if the last word is "music" or "Music"
-     * @param topic identifies youtube video category e.g. Knowledge or Pop music
-     * @return whether topic is categorized as music
+     * @param likedVideoRes json obj of youtube response body 
+     * @return number of total results from json response
      */
-    protected Boolean isMusic(String topic) {
-        String lastWord = topic.substring(topic.lastIndexOf(" ") + 1);
-        return lastWord.equalsIgnoreCase("music");
+    protected int getTotalResults(JsonObject likedVideoRes) {
+        JsonObject pageInfo = likedVideoRes.getAsJsonObject("pageInfo");
+        JsonPrimitive totalResults = pageInfo.getAsJsonPrimitive("totalResults");
+        return totalResults.getAsInt();
     }
 
     /**
-     * parses through youtube liked videos json string,
-     * updates hash map to contain frequency count of each music genre
-     * @param youtubeResBody json response of youtube liked videos
-     * @param genreCount hash map of frequency count of each music genre
-     * @param numVideos maximum number of videos to retrieve
+     * @param likedVideoRes json obj of youtube response body 
+     * @return next page token json primitive
      */
-    protected void updateMusicCount(String youtubeResBody, HashMap<String, Integer> genreCount, int numVideos) {
-        JsonObject jObject = JsonParser.parseString(youtubeResBody).getAsJsonObject();
-        JsonArray videos = jObject.getAsJsonArray("items");
-
-        for (int i = 0; i < videos.size(); i++) {
-            JsonObject video = videos.get(i).getAsJsonObject();
-            JsonObject topicDetails = video.getAsJsonObject("topicDetails");
-
-            if (topicDetails == null) {
-                // Video has no topics so it can't have a music topic
-                continue;
-            }
-
-            JsonArray topicCategories = topicDetails.getAsJsonArray("topicCategories");
-
-            for (int j = 0; j < topicCategories.size(); j++) {
-                // extract music genre out of wikipedia links of topic categories
-                String link = topicCategories.get(j).toString();
-                String topic = link.substring(link.lastIndexOf('/') + 1);
-                topic = topic.replaceAll("\"", "");
-                topic = topic.replaceAll("_", " ");
-
-                if (!isMusic(topic)) {
-                    break;
-                }
-
-                int count = genreCount.containsKey(topic) ? genreCount.get(topic) : 0;
-                genreCount.put(topic, count + 1);
-            }
-        }
-        return;
+    protected JsonPrimitive getNextPageToken(JsonObject likedVideoRes) {
+        JsonPrimitive nextPageToken = likedVideoRes.getAsJsonPrimitive("nextPageToken");
+        return nextPageToken;
```

Comment:
Return `nextPageToken.getAsString()`. This will let you declare `nextPageToken` as a string instead of needing to initialize a json primitive just to unwrap it.